packages/patternfly-4/react-table/src/components/Table/Header.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { Header } from 'reactabular-table';
+import PropTypes from 'prop-types';
+import { TableContext } from './Table';
+import isEqual from 'lodash/isEqual';
+import { SortByDirection } from './SortColumn';
+import { calculateColumns } from './utils/headerUtils';
+
+const propTypes = {
+  className: PropTypes.string,
+  headerRows: PropTypes.array.isRequired,
+  sortBy: PropTypes.shape({
+    index: PropTypes.number,
+    direction: PropTypes.oneOf(Object.values(SortByDirection))
+  }),
+  onCollapse: PropTypes.func,
+  onSort: PropTypes.func,
+  onSelect: PropTypes.func,
+  selectLabeledBy: PropTypes.string,
+  actions: PropTypes.array
+};
+
+const defaultProps = {
+  onCollapse: null,
+  className: '',
+  sortBy: {},
+  onSort: null,
+  onSelect: null
+};
+
+class ContextHeader extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    const { sortBy: nextSortBy, actions: nextAction } = nextProps;
+    const { sortBy, actions, onCollapse, headerRows } = this.props;
+    const updateCheckers = [
```

Comment:
I understand your concern, but this needs to be here so there is not infinite re-render when any kind of data changes. I wanted to keep the API sane, but react-tabular is expecting columns in provider and rows in table body so I had to move columns to table header and pass them up to provider.

But given the controversy over lodash I think that it would be much better to go with different API and pass rows, columns and every config prop to Table provider directly and Header, Body and Footer will consume these props.
```JSX
<Table columns={} rows={} {...props}>
  <Header />
  <Body />
  <Footer />
</Table>
```